vault/src/issue.rs

```diff
@@ -193,25 +193,29 @@ async fn add_new_deposit_key<B: BitcoinCoreApi + Clone + Send + Sync + 'static>(
 /// * `event_channel` - the channel over which to signal events
 /// * `issue_set` - all issue ids observed since vault started
 pub async fn listen_for_issue_requests<B: BitcoinCoreApi + Clone + Send + Sync + 'static>(
-    bitcoin_core: B,
+    btc_rpc: VaultIdManager<B>,
     btc_parachain: InterBtcParachain,
     event_channel: Sender<Event>,
     issue_set: Arc<IssueRequests>,
 ) -> Result<(), ServiceError> {
-    let bitcoin_core = &bitcoin_core;
     let btc_parachain = &btc_parachain;
     let event_channel = &event_channel;
     let issue_set = &issue_set;
+    let btc_rpc = &btc_rpc;
     btc_parachain
         .on_event::<RequestIssueEvent, _, _, _>(
             |event| async move {
                 if &event.vault_id.account_id == btc_parachain.get_account_id() {
+                    let bitcoin_core = match btc_rpc.get_bitcoin_rpc(&event.vault_id).await {
+                        Some(x) => x,
+                        None => return, // event not directed at this vault
```

Comment:
This is a good fix, but in the future please either create a separate PR for unrelated fixes, or if there is a reason not to, then at least mention it in the PR description so we don't miss it in the review. Anyway, let's print an error here - we already checked above that the event is directed at this accountid - if we can't find the bitcoinrpc, it indicates a bug.